src/GoTrueApi.ts

```diff
@@ -70,18 +68,17 @@ export default class GoTrueApi {
       queryParams?: { [key: string]: string }
     }
   ) {
-    const urlParams: string[] = [`provider=${encodeURIComponent(provider)}`]
+    const query = options?.queryParams ? options.queryParams : {}
+    query['provider'] = provider
+
     if (options?.redirectTo) {
-      urlParams.push(`redirect_to=${encodeURIComponent(options.redirectTo)}`)
+      query['redirect_to'] = options.redirectTo
     }
     if (options?.scopes) {
-      urlParams.push(`scopes=${encodeURIComponent(options.scopes)}`)
+      query['scopes'] = options.scopes
     }
-    if (options?.queryParams) {
-      const query = new URLSearchParams(options.queryParams)
-      urlParams.push(`${query}`)
-    }
-    return `${this.url}/authorize?${urlParams.join('&')}`
+    const qs = new URLSearchParams(query).toString()
```

Comment:
This is a small one but seeing as the purpose of this is to clean up code, I like the idea of removing the intermediary `query` object in favour of using the `URLSearchParams` directly. For example:
```javascript
const qs = new URLSearchParams(options?.queryParams).toString()

qs.set('provider', provider)

if (options?.redirectTo) {
  qs.set('redirect_to', options.redirectTo)
}
```